

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>How to Subset a Giotto Object? &mdash; Giotto 1.0 documentation</title>
  

  
  <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
        <script src="../../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="How to Visualize and Save Plots in Giotto?" href="howtovisualizeandsaveplots.html" />
    <link rel="prev" title="Working with the Giotto class: How to create Giotto object with precomputed results?" href="giottoclass.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../index.html" class="icon icon-home"> Giotto
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../general.html">General Information</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../general.html#references">References</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../general.html#other-relevant-information">Other Relevant Information</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html">Getting Started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#description">Description</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#requirements">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#workflow-diagram">Workflow Diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#howtos">HowTos</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#giotto-workflow-analyses"><span class="xref std std-doc">Giotto Workflow Analyses</span></a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html">Giotto Workflow Diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#giotto-workflow-analyses">Giotto Workflow Analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#create-a-giotto-object">1. Create a Giotto Object</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#process-and-filter-a-giotto-object">2. Process and Filter a Giotto Object</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#dimension-reduction">3. Dimension Reduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#cluster-cells-or-spots">4. Cluster cells or spots</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#identify-differentially-expressed-genes">5. Identify differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#annotate-clusters">6. Annotate clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#cell-type-enrichment-or-deconvolution-per-spot">7. Cell-type enrichment or deconvolution per spot</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#create-a-spatial-grid-or-network">8. Create a Spatial grid or Network</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#find-genes-with-a-spatially-coherent-gene-expression-pattern">9. Find genes with a spatially coherent gene expression pattern</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#spatially-coexpressed-genes">10. Identify genes that are spatially co-expressed</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#explore-spatial-domains-with-hmrf">11. Explore spatial domains with HMRF</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#calculate-spatial-cell-cell-interaction-enrichment">12. Calculate spatial cell-cell interaction enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#find-cell-cell-interaction-changed-genes-icg">13. Find cell-cell interaction changed genes (ICG)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#enriched-or-depleted-ligand-receptor-interactions">14. Identify enriched or depleted ligand-receptor interactions in hetero and homo-typic cell interactions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#export-giotto-results-to-use-in-giotto-viewer">15. Export Giotto results to use in Giotto viewer</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../datasets.html">Datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../errorsfaqsandtips.html">FAQs and Solutions to Errors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../errorsfaqsandtips.html#errors-on-macos">Errors on MacOS</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../errorsfaqsandtips.html#errors-on-windows">Errors on Windows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../additionalinformation.html">Changelog</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../additionalinformation.html#contact">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../additionalinformation.html#contribute">Contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="giottoclass.html">Working with the Giotto class: <em>How to create Giotto object with precomputed results?</em></a></li>
<li class="toctree-l1"><a class="reference internal" href="giottoclass.html#the-giotto-class">1. The Giotto Class</a></li>
<li class="toctree-l1"><a class="reference internal" href="giottoclass.html#giotto-slots">2. Giotto Slots</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">How to Subset a Giotto Object?</a></li>
<li class="toctree-l1"><a class="reference internal" href="#the-visium-10x-whole-brain">1. The Visium 10X Whole Brain</a></li>
<li class="toctree-l1"><a class="reference internal" href="#show-only-a-subset-of-clusters">2. Show only a subset of clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="#show-only-a-subset-of-the-clusters-excluding-not-selected-cells">3. Show only a subset of the clusters, excluding not selected cells</a></li>
<li class="toctree-l1"><a class="reference internal" href="#create-spatial-plots-grouped-by-the-leiden-clustering-result">4. Create spatial plots grouped by the leiden clustering result</a></li>
<li class="toctree-l1"><a class="reference internal" href="#create-dimension-plots-grouped-by-the-hmrf-result">5. Create dimension plots grouped by the HMRF result</a></li>
<li class="toctree-l1"><a class="reference internal" href="#create-a-new-giotto-object-by-subsetting-selected-clusters">6. Create a new Giotto object by subsetting selected clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="#create-a-new-giotto-object-by-subsetting-based-on-spatial-coordinates-locations">7. Create a new Giotto object by subsetting based on spatial coordinates/locations</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html">How to Visualize and Save Plots in Giotto?</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html#create-giotto-instructions-and-giotto-object">1. Create Giotto instructions and Giotto object</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html#create-giotto-object">2. Create Giotto Object</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html#working-with-giotto-instructions">3. Working with Giotto Instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html#different-ways-of-saving-a-plot">4. Different Ways of Saving a Plot</a></li>
<li class="toctree-l1"><a class="reference internal" href="testandstoremultipleanayses.html">How to Test and Store Multiple Parameters or Analyses?</a></li>
<li class="toctree-l1"><a class="reference internal" href="testandstoremultipleanayses.html#calculate-highly-variable-genes-in-two-different-manners">1. Calculate highly variable genes in two different manners</a></li>
<li class="toctree-l1"><a class="reference internal" href="testandstoremultipleanayses.html#perform-multiple-pcas">2. Perform Multiple PCAs</a></li>
<li class="toctree-l1"><a class="reference internal" href="testandstoremultipleanayses.html#create-multiple-umaps">3. Create Multiple UMAPs</a></li>
<li class="toctree-l1"><a class="reference internal" href="testandstoremultipleanayses.html#create-multiple-spatial-networks">4. Create Multiple Spatial Networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="testandstoremultipleanayses.html#find-spatial-genes-in-multiple-ways">5. Find Spatial Genes in Multiple Ways</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html">Different ways to visualize your spatial data?</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#functions-specific-to-a-task-or-analysis">1. Functions specific to a task or analysis:</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#functions-that-use-the-spatial-coordinates-of-each-cell-or-spot-they-usually-start-with-spatxxx">2. Functions that use the spatial coordinates of each cell or spot, they usually start with spatXXX</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#functions-that-use-the-dimension-reduction-coordinates-of-each-cell-or-spot-they-usually-start-with-dimxxx">3. Functions that use the dimension reduction coordinates of each cell or spot, they usually start with dimXXX</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#functions-for-co-visualization-that-combine-spat-2-and-dim-3">4. Functions for co-visualization that combine spat (2) and dim (3)</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#both-in-2d-and-3d">5. Both in 2D and 3D</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#ways-to-save-plots">6. Ways to Save Plots</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html">Adding and Working with Images in Giotto: How to Work with Background Images?</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#create-giotto-object">1. Create Giotto Object</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#align-plot">2. Align Plot</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#change-background-of-a-giotto-or-magick-image">3. Change Background of a Giotto or Magick Image</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#add-image-from-scratch">4. Add Image From Scratch</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#example-kidney-analysis">5. Example Kidney Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html">Visualize Spatial Data with Voronoi Plots</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html#start-with-the-analysis-of-the-mini-seqfish-dataset">1. Start with the analysis of the mini seqFISH+ dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html#voronoi-plot-examples">2. Voronoi Plot Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html#creating-neighbors-and-showing-neighbors">3. Creating Neighbors and Showing Neighbors</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html#gene-expression-and-voronoi-plots">4. Gene Expression and Voronoi Plots</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Giotto</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>How to Subset a Giotto Object?</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../../_sources/subsections/tipsandtricks/howtosubsetgiottoobject.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="how-to-subset-a-giotto-object">
<span id="ways-of-subsetting"></span><h1>How to Subset a Giotto Object?<a class="headerlink" href="#how-to-subset-a-giotto-object" title="Permalink to this headline">¶</a></h1>
<p>There are several ways to subset a Giotto object or visualize a subset of the data. Here we will use the already analyzed Visium 10X brain dataset object to:</p>
<ul class="simple">
<li><p>Show how to visualize a subset of cells</p></li>
<li><p>Show how to visualize data by group (example clustering, layers in 3D images, …)</p></li>
<li><p>Subset the dataset based on cell IDs (cluster name(s) or individual cell IDs)</p></li>
<li><p>Subset the dataset based on spatial locations/coordinates</p></li>
</ul>
<p>Figures may not be identical, because examples might have been updated, but the subsetting principles remain the same.</p>
</div>
<div class="section" id="the-visium-10x-whole-brain">
<h1>1. The Visium 10X Whole Brain<a class="headerlink" href="#the-visium-10x-whole-brain" title="Permalink to this headline">¶</a></h1>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">spatDimPlot</span><span class="p">(</span><span class="n">visium_brain</span><span class="p">,</span> <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span>
<span class="n">plot_alignment</span> <span class="o">=</span> <span class="s1">&#39;horizontal&#39;</span><span class="p">,</span> <span class="n">spat_point_size</span> <span class="o">=</span> <span class="mi">2</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/1_whole_brain.png"><img alt="Visium 10X Whole Brain Image" src="../../_images/1_whole_brain.png" style="width: 400px;" /></a>
</div>
<div class="section" id="show-only-a-subset-of-clusters">
<h1>2. Show only a subset of clusters<a class="headerlink" href="#show-only-a-subset-of-clusters" title="Permalink to this headline">¶</a></h1>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">spatDimPlot</span><span class="p">(</span><span class="n">visium_brain</span><span class="p">,</span>
        <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span>  <span class="n">select_cell_groups</span> <span class="o">=</span> <span class="s1">&#39;4&#39;</span><span class="p">,</span>
        <span class="n">plot_alignment</span> <span class="o">=</span> <span class="s1">&#39;horizontal&#39;</span><span class="p">,</span> <span class="n">spat_point_size</span> <span class="o">=</span> <span class="mi">2</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/2_whole_brain_clus4.png"><img alt="Show only a subset of clusters" src="../../_images/2_whole_brain_clus4.png" style="width: 400px;" /></a>
</div>
<div class="section" id="show-only-a-subset-of-the-clusters-excluding-not-selected-cells">
<h1>3. Show only a subset of the clusters, excluding not selected cells<a class="headerlink" href="#show-only-a-subset-of-the-clusters-excluding-not-selected-cells" title="Permalink to this headline">¶</a></h1>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">spatDimPlot</span><span class="p">(</span><span class="n">visium_brain</span><span class="p">,</span>
    <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span>  <span class="n">select_cell_groups</span> <span class="o">=</span> <span class="s1">&#39;4&#39;</span><span class="p">,</span> <span class="n">show_other_cells</span> <span class="o">=</span> <span class="n">F</span><span class="p">,</span>
    <span class="n">plot_alignment</span> <span class="o">=</span> <span class="s1">&#39;horizontal&#39;</span><span class="p">,</span> <span class="n">spat_point_size</span> <span class="o">=</span> <span class="mi">2</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/3_whole_brain_clus4_noOther.png"><img alt="Excluding not selected cells" src="../../_images/3_whole_brain_clus4_noOther.png" style="width: 400px;" /></a>
</div>
<div class="section" id="create-spatial-plots-grouped-by-the-leiden-clustering-result">
<h1>4. Create spatial plots grouped by the leiden clustering result<a class="headerlink" href="#create-spatial-plots-grouped-by-the-leiden-clustering-result" title="Permalink to this headline">¶</a></h1>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">spatPlot</span><span class="p">(</span><span class="n">visium_brain</span><span class="p">,</span>
        <span class="n">group_by</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span> <span class="n">group_by_subset</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="mi">1</span><span class="p">:</span><span class="mi">8</span><span class="p">),</span>
        <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span> <span class="n">cow_n_col</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span>
        <span class="n">point_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">axis_text</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">axis_title</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">legend_text</span> <span class="o">=</span> <span class="mi">6</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/4_whole_brain_by_group_leiden_1.png"><img alt="Leiden group 1" src="../../_images/4_whole_brain_by_group_leiden_1.png" style="width: 400px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">spatPlot</span><span class="p">(</span><span class="n">visium_brain</span><span class="p">,</span>
        <span class="n">group_by</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span> <span class="n">group_by_subset</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="mi">9</span><span class="p">:</span><span class="mi">16</span><span class="p">),</span>
        <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span> <span class="n">cow_n_col</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span>
        <span class="n">point_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">axis_text</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">axis_title</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">legend_text</span> <span class="o">=</span> <span class="mi">6</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/4_whole_brain_by_group_leiden_2.png"><img alt="Leiden group 2" src="../../_images/4_whole_brain_by_group_leiden_2.png" style="width: 400px;" /></a>
</div>
<div class="section" id="create-dimension-plots-grouped-by-the-hmrf-result">
<h1>5. Create dimension plots grouped by the HMRF result<a class="headerlink" href="#create-dimension-plots-grouped-by-the-hmrf-result" title="Permalink to this headline">¶</a></h1>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">spatDimPlot</span><span class="p">(</span><span class="n">visium_brain</span><span class="p">,</span> <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;HMRF_k12_b.1&#39;</span><span class="p">,</span>
          <span class="n">plot_alignment</span> <span class="o">=</span> <span class="s1">&#39;horizontal&#39;</span><span class="p">,</span> <span class="n">spat_point_size</span> <span class="o">=</span> <span class="mi">2</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/5_whole_brain_HMRF.png"><img alt="HMRF Whole Brain" src="../../_images/5_whole_brain_HMRF.png" style="width: 400px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">dimPlot</span><span class="p">(</span><span class="n">visium_brain</span><span class="p">,</span>
        <span class="n">group_by</span> <span class="o">=</span> <span class="s1">&#39;HMRF_k12_b.1&#39;</span><span class="p">,</span> <span class="n">group_by_subset</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="mi">1</span><span class="p">:</span><span class="mi">6</span><span class="p">),</span>
                <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span> <span class="n">cow_n_col</span> <span class="o">=</span> <span class="mi">3</span><span class="p">,</span>
        <span class="n">point_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">axis_text</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">axis_title</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">legend_text</span> <span class="o">=</span> <span class="mi">6</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/5_whole_brain_by_group_HMRF_1.png"><img alt="HMRF Whole Brain by group 1" src="../../_images/5_whole_brain_by_group_HMRF_1.png" style="width: 400px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">dimPlot</span><span class="p">(</span><span class="n">visium_brain</span><span class="p">,</span>
        <span class="n">group_by</span> <span class="o">=</span> <span class="s1">&#39;HMRF_k12_b.1&#39;</span><span class="p">,</span> <span class="n">group_by_subset</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="mi">7</span><span class="p">:</span><span class="mi">14</span><span class="p">),</span>
                <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span> <span class="n">cow_n_col</span> <span class="o">=</span> <span class="mi">3</span><span class="p">,</span>
                <span class="n">point_size</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="n">axis_text</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">axis_title</span> <span class="o">=</span> <span class="mi">6</span><span class="p">,</span> <span class="n">legend_text</span> <span class="o">=</span> <span class="mi">6</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/5_whole_brain_by_group_HMRF_2.png"><img alt="HMRF Whole Brain by group 2" src="../../_images/5_whole_brain_by_group_HMRF_2.png" style="width: 400px;" /></a>
</div>
<div class="section" id="create-a-new-giotto-object-by-subsetting-selected-clusters">
<h1>6. Create a new Giotto object by subsetting selected clusters<a class="headerlink" href="#create-a-new-giotto-object-by-subsetting-selected-clusters" title="Permalink to this headline">¶</a></h1>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>metadata = pDataDT(visium_brain)
subset_cell_IDs = metadata[leiden_clus %in% c(4, 5, 8, 13)]$cell_ID
visium_brain_small = subsetGiotto(visium_brain, cell_ids = subset_cell_IDs)

spatDimPlot(visium_brain_small,
         cell_color = &#39;leiden_clus&#39;,
         plot_alignment = &#39;horizontal&#39;, spat_point_size = 2.5)
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/6_subset_brain.png"><img alt="Subsetting selected clusters" src="../../_images/6_subset_brain.png" style="width: 400px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">spatDimPlot</span><span class="p">(</span><span class="n">visium_brain_small</span><span class="p">,</span>
         <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;nr_genes&#39;</span><span class="p">,</span> <span class="n">color_as_factor</span> <span class="o">=</span> <span class="n">F</span><span class="p">,</span>
         <span class="n">plot_alignment</span> <span class="o">=</span> <span class="s1">&#39;horizontal&#39;</span><span class="p">,</span> <span class="n">spat_point_size</span> <span class="o">=</span> <span class="mf">2.5</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/6_subset_brain_nr_genes.png"><img alt="Subset brain nr" src="../../_images/6_subset_brain_nr_genes.png" style="width: 400px;" /></a>
</div>
<div class="section" id="create-a-new-giotto-object-by-subsetting-based-on-spatial-coordinates-locations">
<h1>7. Create a new Giotto object by subsetting based on spatial coordinates/locations<a class="headerlink" href="#create-a-new-giotto-object-by-subsetting-based-on-spatial-coordinates-locations" title="Permalink to this headline">¶</a></h1>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">DG_subset</span> <span class="o">=</span> <span class="n">subsetGiottoLocs</span><span class="p">(</span><span class="n">visium_brain</span><span class="p">,</span>
                     <span class="n">x_max</span> <span class="o">=</span> <span class="mi">6500</span><span class="p">,</span> <span class="n">x_min</span> <span class="o">=</span> <span class="mi">3000</span><span class="p">,</span>
                     <span class="n">y_max</span> <span class="o">=</span> <span class="o">-</span><span class="mi">2500</span><span class="p">,</span> <span class="n">y_min</span> <span class="o">=</span> <span class="o">-</span><span class="mi">5500</span><span class="p">)</span>

<span class="n">spatDimPlot</span><span class="p">(</span><span class="n">DG_subset</span><span class="p">,</span>
        <span class="n">cell_color</span> <span class="o">=</span> <span class="s1">&#39;leiden_clus&#39;</span><span class="p">,</span>
        <span class="n">plot_alignment</span> <span class="o">=</span> <span class="s1">&#39;horizontal&#39;</span><span class="p">,</span> <span class="n">spat_point_size</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span>
        <span class="n">save_param</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">save_name</span> <span class="o">=</span> <span class="s1">&#39;6_DG_loc_subset&#39;</span><span class="p">,</span> <span class="n">base_height</span> <span class="o">=</span> <span class="mi">4</span><span class="p">))</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/7_DG_loc_subset.png"><img alt="Subset based on spatial coordinates/locations" src="../../_images/7_DG_loc_subset.png" style="width: 400px;" /></a>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="howtovisualizeandsaveplots.html" class="btn btn-neutral float-right" title="How to Visualize and Save Plots in Giotto?" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="giottoclass.html" class="btn btn-neutral float-left" title="Working with the Giotto class: How to create Giotto object with precomputed results?" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, Ruben Dries, Qian Zhu, Huipeng Li, Rui Dong, Guo-Cheng Yuan.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>