

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>How to Test and Store Multiple Parameters or Analyses? &mdash; Giotto 1.0 documentation</title>
  

  
  <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
        <script src="../../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Different ways to visualize your spatial data?" href="waystovisualizespatialdata.html" />
    <link rel="prev" title="How to Visualize and Save Plots in Giotto?" href="howtovisualizeandsaveplots.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../index.html" class="icon icon-home"> Giotto
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../general.html">General Information</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../general.html#references">References</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../general.html#other-relevant-information">Other Relevant Information</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html">Getting Started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#description">Description</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#requirements">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#workflow-diagram">Workflow Diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#howtos">HowTos</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html#giotto-workflow-analyses"><span class="xref std std-doc">Giotto Workflow Analyses</span></a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html">Giotto Workflow Diagram</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#giotto-workflow-analyses">Giotto Workflow Analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#create-a-giotto-object">1. Create a Giotto Object</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#process-and-filter-a-giotto-object">2. Process and Filter a Giotto Object</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#dimension-reduction">3. Dimension Reduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#cluster-cells-or-spots">4. Cluster cells or spots</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#identify-differentially-expressed-genes">5. Identify differentially expressed genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#annotate-clusters">6. Annotate clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#cell-type-enrichment-or-deconvolution-per-spot">7. Cell-type enrichment or deconvolution per spot</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#create-a-spatial-grid-or-network">8. Create a Spatial grid or Network</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#find-genes-with-a-spatially-coherent-gene-expression-pattern">9. Find genes with a spatially coherent gene expression pattern</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#spatially-coexpressed-genes">10. Identify genes that are spatially co-expressed</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#explore-spatial-domains-with-hmrf">11. Explore spatial domains with HMRF</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#calculate-spatial-cell-cell-interaction-enrichment">12. Calculate spatial cell-cell interaction enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#find-cell-cell-interaction-changed-genes-icg">13. Find cell-cell interaction changed genes (ICG)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#enriched-or-depleted-ligand-receptor-interactions">14. Identify enriched or depleted ligand-receptor interactions in hetero and homo-typic cell interactions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../giottoworkflowanalyses.html#export-giotto-results-to-use-in-giotto-viewer">15. Export Giotto results to use in Giotto viewer</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../datasets.html">Datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../errorsfaqsandtips.html">FAQs and Solutions to Errors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../errorsfaqsandtips.html#errors-on-macos">Errors on MacOS</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../errorsfaqsandtips.html#errors-on-windows">Errors on Windows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../additionalinformation.html">Changelog</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../additionalinformation.html#contact">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../additionalinformation.html#contribute">Contribute</a></li>
<li class="toctree-l1"><a class="reference internal" href="giottoclass.html">Working with the Giotto class: <em>How to create Giotto object with precomputed results?</em></a></li>
<li class="toctree-l1"><a class="reference internal" href="giottoclass.html#the-giotto-class">1. The Giotto Class</a></li>
<li class="toctree-l1"><a class="reference internal" href="giottoclass.html#giotto-slots">2. Giotto Slots</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtosubsetgiottoobject.html">How to Subset a Giotto Object?</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtosubsetgiottoobject.html#the-visium-10x-whole-brain">1. The Visium 10X Whole Brain</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtosubsetgiottoobject.html#show-only-a-subset-of-clusters">2. Show only a subset of clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtosubsetgiottoobject.html#show-only-a-subset-of-the-clusters-excluding-not-selected-cells">3. Show only a subset of the clusters, excluding not selected cells</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtosubsetgiottoobject.html#create-spatial-plots-grouped-by-the-leiden-clustering-result">4. Create spatial plots grouped by the leiden clustering result</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtosubsetgiottoobject.html#create-dimension-plots-grouped-by-the-hmrf-result">5. Create dimension plots grouped by the HMRF result</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtosubsetgiottoobject.html#create-a-new-giotto-object-by-subsetting-selected-clusters">6. Create a new Giotto object by subsetting selected clusters</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtosubsetgiottoobject.html#create-a-new-giotto-object-by-subsetting-based-on-spatial-coordinates-locations">7. Create a new Giotto object by subsetting based on spatial coordinates/locations</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html">How to Visualize and Save Plots in Giotto?</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html#create-giotto-instructions-and-giotto-object">1. Create Giotto instructions and Giotto object</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html#create-giotto-object">2. Create Giotto Object</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html#working-with-giotto-instructions">3. Working with Giotto Instructions</a></li>
<li class="toctree-l1"><a class="reference internal" href="howtovisualizeandsaveplots.html#different-ways-of-saving-a-plot">4. Different Ways of Saving a Plot</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">How to Test and Store Multiple Parameters or Analyses?</a></li>
<li class="toctree-l1"><a class="reference internal" href="#calculate-highly-variable-genes-in-two-different-manners">1. Calculate highly variable genes in two different manners</a></li>
<li class="toctree-l1"><a class="reference internal" href="#perform-multiple-pcas">2. Perform Multiple PCAs</a></li>
<li class="toctree-l1"><a class="reference internal" href="#create-multiple-umaps">3. Create Multiple UMAPs</a></li>
<li class="toctree-l1"><a class="reference internal" href="#create-multiple-spatial-networks">4. Create Multiple Spatial Networks</a></li>
<li class="toctree-l1"><a class="reference internal" href="#find-spatial-genes-in-multiple-ways">5. Find Spatial Genes in Multiple Ways</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html">Different ways to visualize your spatial data?</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#functions-specific-to-a-task-or-analysis">1. Functions specific to a task or analysis:</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#functions-that-use-the-spatial-coordinates-of-each-cell-or-spot-they-usually-start-with-spatxxx">2. Functions that use the spatial coordinates of each cell or spot, they usually start with spatXXX</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#functions-that-use-the-dimension-reduction-coordinates-of-each-cell-or-spot-they-usually-start-with-dimxxx">3. Functions that use the dimension reduction coordinates of each cell or spot, they usually start with dimXXX</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#functions-for-co-visualization-that-combine-spat-2-and-dim-3">4. Functions for co-visualization that combine spat (2) and dim (3)</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#both-in-2d-and-3d">5. Both in 2D and 3D</a></li>
<li class="toctree-l1"><a class="reference internal" href="waystovisualizespatialdata.html#ways-to-save-plots">6. Ways to Save Plots</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html">Adding and Working with Images in Giotto: How to Work with Background Images?</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#create-giotto-object">1. Create Giotto Object</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#align-plot">2. Align Plot</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#change-background-of-a-giotto-or-magick-image">3. Change Background of a Giotto or Magick Image</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#add-image-from-scratch">4. Add Image From Scratch</a></li>
<li class="toctree-l1"><a class="reference internal" href="workingwithimages.html#example-kidney-analysis">5. Example Kidney Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html">Visualize Spatial Data with Voronoi Plots</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html#start-with-the-analysis-of-the-mini-seqfish-dataset">1. Start with the analysis of the mini seqFISH+ dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html#voronoi-plot-examples">2. Voronoi Plot Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html#creating-neighbors-and-showing-neighbors">3. Creating Neighbors and Showing Neighbors</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualizewithvoronoiplots.html#gene-expression-and-voronoi-plots">4. Gene Expression and Voronoi Plots</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Giotto</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>How to Test and Store Multiple Parameters or Analyses?</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../../_sources/subsections/tipsandtricks/testandstoremultipleanayses.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="how-to-test-and-store-multiple-parameters-or-analyses">
<span id="test-and-store"></span><h1>How to Test and Store Multiple Parameters or Analyses?<a class="headerlink" href="#how-to-test-and-store-multiple-parameters-or-analyses" title="Permalink to this headline">¶</a></h1>
<p>The default Giotto workflow is similar to other scRNA-seq workflows and does not require you to provide a custom name for each analysis (e.g. PCA, UMAP, …), but running an analysis twice will overwrite the previous results with a warning. However, there are situations where being able to run and store multiple analyses can be advantageous:</p>
<ul class="simple">
<li><p>Test multiple parameters for a single analysis</p></li>
<li><p>Test multiple combinations across functions (See Example: <strong>hvg-&gt;pca-&gt;umap</strong>)</p></li>
<li><p>Use different output results as input for downstream analyses (See Example:  <strong>spatial genes</strong>)</p></li>
</ul>
<a class="reference internal image-reference" href="../../_images/ppt_store_multiple_analyses2.png"><img alt="Multiple Analysis" class="align-center" src="../../_images/ppt_store_multiple_analyses2.png" style="width: 600px;" /></a>
<p>We will use the seqFish+ somatosensory cortex as an example dataset after creating and processing a Giotto object.</p>
</div>
<div class="section" id="calculate-highly-variable-genes-in-two-different-manners">
<h1>1. Calculate highly variable genes in two different manners<a class="headerlink" href="#calculate-highly-variable-genes-in-two-different-manners" title="Permalink to this headline">¶</a></h1>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># using the loess method</span>
<span class="n">VC_test</span> <span class="o">&lt;-</span> <span class="n">calculateHVG</span><span class="p">(</span><span class="n">gobject</span> <span class="o">=</span> <span class="n">VC_test</span><span class="p">,</span>
                    <span class="n">method</span> <span class="o">=</span> <span class="s1">&#39;cov_loess&#39;</span><span class="p">,</span> <span class="n">difference_in_cov</span> <span class="o">=</span> <span class="mf">0.1</span><span class="p">,</span>
                    <span class="n">HVGname</span> <span class="o">=</span> <span class="s1">&#39;loess_hvg&#39;</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/1_loess_hvg.png"><img alt="Loess" src="../../_images/1_loess_hvg.png" style="width: 300px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># using the expression groups method</span>
<span class="n">VC_test</span> <span class="o">&lt;-</span> <span class="n">calculateHVG</span><span class="p">(</span><span class="n">gobject</span> <span class="o">=</span> <span class="n">VC_test</span>
                    <span class="p">,</span> <span class="n">method</span> <span class="o">=</span> <span class="s1">&#39;cov_group&#39;</span><span class="p">,</span> <span class="n">zscore_threshold</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span>
                    <span class="n">HVGname</span> <span class="o">=</span> <span class="s1">&#39;group_hvg&#39;</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/1_group_hvg.png"><img alt="Group" src="../../_images/1_group_hvg.png" style="width: 300px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span># compare the highly variable genes between two methods
gene_metadata = fDataDT(VC_test)
mytable = table(loess = gene_metadata$loess_hvg, group = gene_metadata$group_hvg)
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/1_hvg_table.png"><img alt="Group" src="../../_images/1_hvg_table.png" style="width: 200px;" /></a>
</div>
<div class="section" id="perform-multiple-pcas">
<h1>2. Perform Multiple PCAs<a class="headerlink" href="#perform-multiple-pcas" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p>Using the 2 different HVG sets (loess_genes and group_genes)</p></li>
<li><p>Store PCA results using custom names (‘pca_loess’ and ‘pca_group’)</p></li>
<li><p>Plot PCA results</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>## 2. PCA ##
# pca with genes from loess
loess_genes = gene_metadata[loess_hvg == &#39;yes&#39;]$gene_ID
VC_test &lt;- runPCA(gobject = VC_test, genes_to_use = loess_genes, name = &#39;pca_loess&#39;, scale_unit = F)
plotPCA(gobject = VC_test, dim_reduction_name = &#39;pca_loess&#39;)
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/2_loess_pca.png"><img alt="Group" src="../../_images/2_loess_pca.png" style="width: 300px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span># pca with genes from group
group_genes = gene_metadata[group_hvg == &#39;yes&#39;]$gene_ID
VC_test &lt;- runPCA(gobject = VC_test, genes_to_use = group_genes, name = &#39;pca_group&#39;, scale_unit = F)
plotPCA(gobject = VC_test, dim_reduction_name = &#39;pca_group&#39;)
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/2_group_pca.png"><img alt="Group" src="../../_images/2_group_pca.png" style="width: 300px;" /></a>
</div>
<div class="section" id="create-multiple-umaps">
<h1>3. Create Multiple UMAPs<a class="headerlink" href="#create-multiple-umaps" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p>Using the 2 different PCA results (‘pca_loess’ and ‘pca_group’)</p></li>
<li><p>Store UMAP results using custom names (‘umap_loess’ and ‘umap_group’)</p></li>
<li><p>Plot UMAP results</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">## 3. UMAP ##</span>
  <span class="n">VC_test</span> <span class="o">&lt;-</span> <span class="n">runUMAP</span><span class="p">(</span><span class="n">VC_test</span><span class="p">,</span> <span class="n">dim_reduction_to_use</span> <span class="o">=</span> <span class="s1">&#39;pca&#39;</span><span class="p">,</span> <span class="n">dim_reduction_name</span> <span class="o">=</span> <span class="s1">&#39;pca_loess&#39;</span><span class="p">,</span>
                 <span class="n">name</span> <span class="o">=</span> <span class="s1">&#39;umap_loess&#39;</span><span class="p">,</span> <span class="n">dimensions_to_use</span> <span class="o">=</span> <span class="mi">1</span><span class="p">:</span><span class="mi">30</span><span class="p">)</span>
  <span class="n">plotUMAP</span><span class="p">(</span><span class="n">gobject</span> <span class="o">=</span> <span class="n">VC_test</span><span class="p">,</span> <span class="n">dim_reduction_name</span> <span class="o">=</span> <span class="s1">&#39;umap_loess&#39;</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/3_loess_umap.png"><img alt="Group" src="../../_images/3_loess_umap.png" style="width: 300px;" /></a>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">VC_test</span> <span class="o">&lt;-</span> <span class="n">runUMAP</span><span class="p">(</span><span class="n">VC_test</span><span class="p">,</span> <span class="n">dim_reduction_to_use</span> <span class="o">=</span> <span class="s1">&#39;pca&#39;</span><span class="p">,</span> <span class="n">dim_reduction_name</span> <span class="o">=</span> <span class="s1">&#39;pca_group&#39;</span><span class="p">,</span>
               <span class="n">name</span> <span class="o">=</span> <span class="s1">&#39;umap_group&#39;</span><span class="p">,</span> <span class="n">dimensions_to_use</span> <span class="o">=</span> <span class="mi">1</span><span class="p">:</span><span class="mi">30</span><span class="p">)</span>
<span class="n">plotUMAP</span><span class="p">(</span><span class="n">gobject</span> <span class="o">=</span> <span class="n">VC_test</span><span class="p">,</span> <span class="n">dim_reduction_name</span> <span class="o">=</span> <span class="s1">&#39;umap_group&#39;</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/3_group_umap.png"><img alt="Group" src="../../_images/3_group_umap.png" style="width: 300px;" /></a>
</div>
<div class="section" id="create-multiple-spatial-networks">
<h1>4. Create Multiple Spatial Networks<a class="headerlink" href="#create-multiple-spatial-networks" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p>Create spatial with multiple k’s and other parameters (k=5, k=10, k=100 &amp; maximum_distance=200)</p></li>
<li><p>Subset field 1</p></li>
<li><p>Visualize network on field 1 (‘spatial_network’, ‘large_network’, ‘distance_work’)</p></li>
</ul>
<p><em>Spatial Network</em></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>## 4. spatial network
VC_test &lt;- createSpatialNetwork(gobject = VC_test, method = &#39;kNN&#39;, k = 5) # standard name: &#39;spatial_network&#39;
VC_test &lt;- createSpatialNetwork(gobject = VC_test, method = &#39;kNN&#39;, k = 10, name = &#39;large_network&#39;)  VC_test &lt;- createSpatialNetwork(gobject = VC_test, method = &#39;kNN&#39;, k = 100, maximum_distance_knn = 200, minimum_k = 2, name = &#39;distance_network&#39;)

## visualize different spatial networks on first field (~ layer 1)
cell_metadata = pDataDT(VC_test)
field1_ids = cell_metadata[Field_of_View == 0]$cell_ID
subVC_test = subsetGiotto(VC_test, cell_ids = field1_ids)

spatPlot(gobject = subVC_test, show_network = T,
         network_color = &#39;blue&#39;, spatial_network_name = &#39;spatial_network&#39;)
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/4_spatial_network.png"><img alt="Group" src="../../_images/4_spatial_network.png" style="width: 300px;" /></a>
<p><em>Large Network</em></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">spatPlot</span><span class="p">(</span><span class="n">gobject</span> <span class="o">=</span> <span class="n">subVC_test</span><span class="p">,</span> <span class="n">show_network</span> <span class="o">=</span> <span class="n">T</span><span class="p">,</span>
         <span class="n">network_color</span> <span class="o">=</span> <span class="s1">&#39;blue&#39;</span><span class="p">,</span> <span class="n">spatial_network_name</span> <span class="o">=</span> <span class="s1">&#39;large_network&#39;</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/4_large_network.png"><img alt="Group" src="../../_images/4_large_network.png" style="width: 300px;" /></a>
<p><em>Distance Network</em></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">spatPlot</span><span class="p">(</span><span class="n">gobject</span> <span class="o">=</span> <span class="n">subVC_test</span><span class="p">,</span> <span class="n">show_network</span> <span class="o">=</span> <span class="n">T</span><span class="p">,</span>
         <span class="n">network_color</span> <span class="o">=</span> <span class="s1">&#39;blue&#39;</span><span class="p">,</span> <span class="n">spatial_network_name</span> <span class="o">=</span> <span class="s1">&#39;distance_network&#39;</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/4_distance_network.png"><img alt="Group" src="../../_images/4_distance_network.png" style="width: 300px;" /></a>
</div>
<div class="section" id="find-spatial-genes-in-multiple-ways">
<h1>5. Find Spatial Genes in Multiple Ways<a class="headerlink" href="#find-spatial-genes-in-multiple-ways" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p>Use the different spatial networks as input to identify spatial genes with the rank method</p></li>
<li><p>Visualize top spatial genes for 2 methods</p></li>
</ul>
<p><em>Large Network Spatial Genes</em></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>## 5. spatial genes
 # the provided spatial_network_name can be given to downstream analyses

 # spatial genes based on large network
 ranktest_large = binSpect(VC_test,
                           subset_genes = loess_genes,
                           bin_method = &#39;rank&#39;,
                           spatial_network_name = &#39;large_network&#39;)


 spatGenePlot(VC_test,
              expression_values = &#39;scaled&#39;,
              genes = ranktest_large$genes[1:6], cow_n_col = 2, point_size = 1,
              genes_high_color = &#39;red&#39;, genes_mid_color = &#39;white&#39;, genes_low_color = &#39;darkblue&#39;, midpoint = 0)
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/5_large_network_genes.png"><img alt="Group" src="../../_images/5_large_network_genes.png" style="width: 300px;" /></a>
<p><em>Distance Network Spatial Genes</em></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span># spatial genes based on distance network
ranktest_dist = binSpect(VC_test,
                         subset_genes = loess_genes,
                         bin_method = &#39;rank&#39;,
                         spatial_network_name = &#39;distance_network&#39;)
spatGenePlot(VC_test,
             expression_values = &#39;scaled&#39;,
             genes = ranktest_dist$genes[1:6], cow_n_col = 2, point_size = 1,
             genes_high_color = &#39;red&#39;, genes_mid_color = &#39;white&#39;, genes_low_color = &#39;darkblue&#39;, midpoint = 0)
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/5_large_distance_network_genes.png"><img alt="Group" src="../../_images/5_large_distance_network_genes.png" style="width: 300px;" /></a>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="waystovisualizespatialdata.html" class="btn btn-neutral float-right" title="Different ways to visualize your spatial data?" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="howtovisualizeandsaveplots.html" class="btn btn-neutral float-left" title="How to Visualize and Save Plots in Giotto?" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, Ruben Dries, Qian Zhu, Huipeng Li, Rui Dong, Guo-Cheng Yuan.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>