General 
==========================================================

The Giotto package consists of two modules, Giotto Analyzer and Viewer (see `www.spatialgiotto.com`_), which provide tools to process, analyze and visualize single-cell spatial expression data. The underlying framework is generalizable to virtually all currently available spatial datasets. We recently demonstrated the general applicability on 10 different datasets created by 9 different state-of-the-art spatial technologies, including in situ hybridization (seqFISH+, merFISH, osmFISH), sequencing (Slide-seq, Visium, STARmap) and imaging-based multiplexing/proteomics (CyCIF, MIBI, CODEX). These technologies differ in terms of resolution (single cell vs multiple cells), spatial dimension (2D vs 3D), molecular modality (protein vs RNA), and throughput (number of cells and genes). More information and documentation about the latest version of Giotto Analyzer can be found at https://rubd.github.io/Giotto_site/ (**URL CHANGE**).

.. _www.spatialgiotto.com: http://spatialgiotto.rc.fas.harvard.edu

.. image:: _build/html/_static/overview_datasets.png
    :width: 400

References
==============================================================
* `Dries, R., Zhu, Q. et al. Giotto, a toolbox for integrative analysis and visualization of spatial expression data. bioRxiv 701680 (2019). doi:10.1101/701680.`_
* `Eng, C.-H. L. et al. Transcriptome-scale super-resolved imaging in tissues by RNA seqFISH+. Nature 1 (2019). doi:10.1038/s41586-019-1049-y.`_.
* `Zhu, Q., Shah, S., Dries, R., Cai, L. & Yuan, G.-C. Identification of spatially associated subpopulations by combining scRNAseq and sequential fluorescence in situ hybridization data. Nature Biotechnology (2018). doi:10.1038/nbt.4260.`_


.. _Dries, R., Zhu, Q. et al. Giotto, a toolbox for integrative analysis and visualization of spatial expression data. bioRxiv 701680 (2019). doi:10.1101/701680.: https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02286-2
.. _Eng, C.-H. L. et al. Transcriptome-scale super-resolved imaging in tissues by RNA seqFISH+. Nature 1 (2019). doi:10.1038/s41586-019-1049-y.: https://www.nature.com/articles/s41586-019-1049-y
.. _Zhu, Q., Shah, S., Dries, R., Cai, L. & Yuan, G.-C. Identification of spatially associated subpopulations by combining scRNAseq and sequential fluorescence in situ hybridization data. Nature Biotechnology (2018). doi:10.1038/nbt.4260.: https://www.nature.com/articles/nbt.4260


Other Relevant Information
==============================================================

Links
--------------------------------------------------------------
* `Browse source code`_ 
* `Report a bug`_

.. _Browse source code: https://github.com/RubD/Giotto/ 
.. _Report a bug: https://github.com/RubD/Giotto/issues

License
---------------------------------------------------------------
`MIT`_ | `File License`_ 

.. _MIT: https://opensource.org/licenses/mit-license.php
.. _File License: https://rubd.github.io/Giotto_site/LICENSE-text.html

`Citation`_
.. _Citation: https://rubd.github.io/Giotto_site/authors.html
---------------------------------------------------------------


Developers 
--------------------------------------------------------------

+---------------+---------------------+
|`Ruben Dries`_ |*Author, Maintainer* | 
+---------------+---------------------+
| Qian Zhu      |*Author*             |
+---------------+---------------------+
| Huipeng Li    |*Author*             |
+---------------+---------------------+
| Rui Dong      |*Author*             |
+---------------+---------------------+
| Guo-Cheng Yuan|*Author*             |
+---------------+---------------------+

.. _Ruben Dries: https://www.drieslab.com
